### PR TITLE
penalty and cooldown data is not erased with retirement

### PR DESCRIPTION
**Pull request by Anze1m**

### CODE REVIEW

File: contracts/Storage/AtlasStakeStore.sol [hunk 1]
```diff
@@ -79,7 +79,10 @@ contract AtlasStakeStore is Base {
         require(refundAddress != address(0));
         require(!isShelteringAny(node));
         uint amount = stakes[node].amount;
-        delete stakes[node];
+        stakes[node].initialAmount = 0;
+        stakes[node].amount = 0;
+        stakes[node].storageLimit = 0;
+        stakes[node].storageUsed = 0;
```
Comment (spherefoundry) [inline contracts/Storage/AtlasStakeStore.sol]: But still calling `depositStake` will zero them out 🤔

Comment (Anze1m) [reply contracts/Storage/AtlasStakeStore.sol]: Good point 😁

File: contracts/Storage/AtlasStakeStore.sol [hunk 2]
```diff
@@ -70,7 +70,10 @@ contract AtlasStakeStore is Base {
     function depositStake(address _who, uint _storageLimit) public payable onlyContextInternalCalls {
         require(!isStaking(_who));
 
-        stakes[_who] = Stake(msg.value, msg.value, _storageLimit, 0, 0, 0, 0);
+        stakes[_who].initialAmount = msg.value;
+        stakes[_who].amount = msg.value;
+        stakes[_who].storageLimit = _storageLimit;
+        stakes[_who].storageUsed = 0;
```
Comment (spherefoundry) [inline contracts/Storage/AtlasStakeStore.sol]: test?

Comment (Anze1m) [reply contracts/Storage/AtlasStakeStore.sol]:
```

    it('penalty and cooldown data is not erased', async () => {
      await setPenaltyHistory(shelterer, examplePenaltyCount, examplePenaltyTimestamp);
      await updateLastChallengeResolvedSequenceNumber(shelterer, exampleChallengeSequenceNumber);

      await releaseStake(shelterer, otherAddress);

      const penaltiesHistory = await getPenaltiesHistory(shelterer);
      expect(penaltiesHistory.penaltiesCount).to.be.equal(examplePenaltyCount);
      expect(penaltiesHistory.lastPenaltyTime).to.be.equal(examplePenaltyTimestamp);
      expect(await getLastChallengeResolvedSequenceNumber(shelterer)).to.be.equal(exampleChallengeSequenceNumber);
    });

```

Comment (Anze1m) [reply contracts/Storage/AtlasStakeStore.sol]: added one more sanity check

**Pull request merged**